moveit_ros/planning/planning_request_adapter_plugins/src/add_ruckig_traj_smoothing.cpp

```diff
@@ -64,17 +64,17 @@ class AddRuckigTrajectorySmoothing : public planning_request_adapter::PlanningRe
                     const planning_interface::MotionPlanRequest& req, planning_interface::MotionPlanResponse& res,
                     std::vector<std::size_t>& /*added_path_index*/) const override
   {
-    bool result = planner(planning_scene, req, res);
-    if (result && res.trajectory_)
+    moveit::core::MoveItErrorCode moveit_code = planner(planning_scene, req, res);
+    if (bool(moveit_code) && res.trajectory_)
     {
       if (!smoother_.applySmoothing(*res.trajectory_, req.max_velocity_scaling_factor,
                                     req.max_acceleration_scaling_factor))
       {
-        result = false;
+        moveit_code = moveit::core::MoveItErrorCode::FAILURE;
       }
     }
 
-    return result;
+    return bool(moveit_code);
```

Comment:
If you pass back the error code here instead of bool, and then pass it through in `callAdapter()`, that should be all you need. The termination condition is already built in, since smoothing is only applied in the happy path.

Reply:
I will try it. I think it's going to require another large set of changes.

After this request, I'm going to claim **scope creep** because our client needs to get this merged soon.

Reply:
Done